File: 2022/typescript/day03/solution.test.ts
import { part1, part2 } from "./solution";

describe("solution", () => {
  it("solves part1", () => {
    expect(part1("day03/sample")).toEqual(157);
    expect(part1("day03/input")).toEqual(8394);
  });

  it("solves part2", () => {
    expect(part2("day03/sample")).toEqual(70);
    expect(part2("day03/input")).toEqual(2413);
  });
});

File: 2015/java/aoc/src/test/java/aoc2015/day08/SolutionTest.java
package aoc.day08;

import aoc.testutils.InputUtils;
import org.junit.Assert;
import org.junit.Test;

public class SolutionTest {
    @Test
    public void solvesPart1() {
        Assert.assertEquals(12, Solution.part1(InputUtils.readInputAsStrings(this, "sample")));
        Assert.assertEquals(1333, Solution.part1(InputUtils.readInputAsStrings(this)));
    }

    @Test
    public void solvesPart2() {
        Assert.assertEquals(19, Solution.part2(InputUtils.readInputAsStrings(this, "sample")));
        Assert.assertEquals(2046, Solution.part2(InputUtils.readInputAsStrings(this)));
    }
}
